addon/session-stores/local-storage.js

```diff
@@ -39,24 +39,30 @@ export default BaseStore.extend({
 
     @method persist
     @param {Object} data The data to persist
+    @return {Ember.RSVP.Promise} The promise object persisting the data in the store.
     @public
   */
   persist(data) {
-    data = JSON.stringify(data || {});
-    localStorage.setItem(this.key, data);
-    this._lastData = this.restore();
+    return new RSVP.Promise((resolve, reject) => {
+      data = JSON.stringify(data || {});
+      localStorage.setItem(this.key, data);
+      this.restore().then((restoredContent) => {
+        this._lastData = restoredContent;
+        resolve();
+      }, reject);
```

Comment:
Same as above - the data would already have been written when `restore` rejects.